modules/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSasTokenTests.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.repositories.azure;
+
+import org.elasticsearch.common.UUIDs;
+import org.elasticsearch.common.blobstore.BlobContainer;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.Streams;
+import org.elasticsearch.common.settings.MockSecureSettings;
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.InputStream;
+import java.util.Locale;
+
+import static org.elasticsearch.repositories.azure.AzureStorageSettings.ACCOUNT_SETTING;
+import static org.elasticsearch.repositories.azure.AzureStorageSettings.SAS_TOKEN_SETTING;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThan;
+
+@SuppressForbidden(reason = "use a http server")
+public class AzureSasTokenTests extends AbstractAzureServerTestCase {
+    public void testSasTokenIsUsedAsProvidedInSettings() throws Exception {
+        final int maxRetries = randomIntBetween(1, 5);
+        final byte[] bytes = randomBlobContent();
+
+        final String clientName = randomAlphaOfLength(5).toLowerCase(Locale.ROOT);
+        final MockSecureSettings secureSettings = new MockSecureSettings();
+        secureSettings.setString(ACCOUNT_SETTING.getConcreteSettingForNamespace(clientName).getKey(), "account");
+        final String sasToken;
+        if (randomBoolean()) {
+            sasToken = "?sv=2018-11-09&spr=https&st=2021-09-21T13%3A00%3A07Z&se=2071-09-21T13%3A00%3A07Z"
+                + "&sr=c&sp=racwdl&sig=4%2Fak3ibKW%2FXILJI%2B8mInVaLiDw8n3Es%2FQbTSiG3LOt0%3D";
+        } else {
+            sasToken = "sv=2018-11-09&spr=https&st=2021-09-21T13%3A00%3A07Z&se=2071-09-21T13%3A00%3A07Z&"
+                + "sr=c&sp=racwdl&sig=4%2Fak3ibKW%2FXILJI%2B8mInVaLiDw8n3Es%2FQbTSiG3LOt0%3D";
+        }
+        secureSettings.setString(SAS_TOKEN_SETTING.getConcreteSettingForNamespace(clientName).getKey(), sasToken);
+
+        httpServer.createContext("/account/container/sas_test", exchange -> {
+            final var queryParams = exchange.getRequestURI().getRawQuery();
+            if (sasToken.startsWith("?")) {
+                assertThat(queryParams, is(equalTo(sasToken.substring(1))));
+            } else {
+                assertThat(queryParams, is(equalTo(sasToken)));
+            }
+
+            Streams.readFully(exchange.getRequestBody());
```

Comment:
Do we need to read the body if only care about the request method?

Reply:
Yes, to ensure that we read the entire request.